platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/LongListHeap.java

```diff
@@ -87,23 +92,24 @@ public LongListHeap(final Path file, final Configuration configuration) throws I
 
     /** {@inheritDoc} */
     @Override
-    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
-        // read data
-        final int numOfArrays = calculateNumberOfChunks(size());
-        final ByteBuffer buffer = allocateDirect(memoryChunkSize);
-        buffer.order(ByteOrder.nativeOrder());
-        for (int i = 0; i < numOfArrays; i++) {
-            final AtomicLongArray atomicLongArray = new AtomicLongArray(numLongsPerChunk);
-            buffer.clear();
-            MerkleDbFileUtils.completelyRead(fileChannel, buffer);
-            buffer.flip();
-            int index = 0;
-            while (buffer.remaining() > 0) {
-                atomicLongArray.set(index, buffer.getLong());
-                index++;
-            }
-            chunkList.set(i, atomicLongArray);
+    protected AtomicLongArray readChunkData(FileChannel fileChannel, int chunkIndex, int startIndex)
+            throws IOException {
+        AtomicLongArray chunk = createChunk();
+
+        int startByteOffset = startIndex * Long.BYTES;
+
+        TEMP_READ_BUFFER.position(0);
+        TEMP_READ_BUFFER.limit(memoryChunkSize - startByteOffset);
```

Comment:
In `LongListDisk`, buffer position is set to `startOffset` and limit to `memoryChunkSize` (should be changed to `endOffset`). Here position is `0` and limit is `memoryChunkSize - startByteOffset` (should be changed to `endOffset - startOffset`). I propose to use the same approach in both classes to make code more consistent and easier to read.